### PR TITLE
Fix/navlink active

### PR DESCRIPTION
**Pull request by sunilsabatp**
Fixes https://planetde.slack.com/archives/C02393H3GDC/p1646898291140889

Changes in this pull request:
- Fix all nav link being active on sub link item being selected

### CODE REVIEW

**Comment by sagararyal** [comment]
- [ ] Translation is broken on this PR

Screen Shot 2022-03-11 at 4 03 43 PM

**Comment by norbertschuler** [comment]
- [ ] Is it possible to keep the menu item expanded where the subitem got selected? Currently for me it get's closed after selecting the sub menu item:

Bildschirmfoto 2022-03-12 um 14 57 49

**Comment by norbertschuler** [comment]
> * [ ]  Translation is broken on this PR

They is a comma missing in the `me.json` file with the English translations. Rebasing this PR onto develop would fix this.

**Review by norbertschuler — CHANGES_REQUESTED** [review]
Please rebase on `develop`.

File: src/features/common/Layout/UserLayout/UserLayout.tsx [hunk 1]
```diff
@@ -169,21 +169,24 @@ function UserLayout(props: any): ReactElement {
...
         if (router.router?.asPath === link.path) {
-          setactiveLink(link.path);
+            setactiveLink(link.path);
         } else if (link.subMenu && link.subMenu.length > 0) {
           const subMenuItem = link.subMenu.find((subMenuItem: any) => {
             return subMenuItem.path === router.router?.asPath;
-          });
+          })
           if (subMenuItem) {
+           if (subMenuItem && typeof subMenuItem !== 'undefined') {
```
Comment (ankitecd) [inline src/features/common/Layout/UserLayout/UserLayout.tsx:183]: First condition also checks if the the property is `undefined` or not. How does the second condition works here?

File: src/features/common/Layout/UserLayout/UserLayout.tsx [hunk 2]
```diff
@@ -218,6 +221,7 @@ function UserLayout(props: any): ReactElement {
               </div>
             </div>
             {navLinks.map((link: any, index: any) => (
+              <>
```
Comment (ankitecd) [inline src/features/common/Layout/UserLayout/UserLayout.tsx:224]: I don't think there is need to have fragment as there is only one component i.e `NavLink`

File: src/features/common/Layout/UserLayout/UserLayout.tsx [hunk 3]
```diff
@@ -338,12 +350,11 @@ function NavLink({
...
   }
 
-  const [isSubMenuActive, setisSubMenuActive] = React.useState(false);
-
   return (
     <div key={link.title} className={styles.navlinkMenu}>
+      
       <div
-        className={`${styles.navlink} ${activeLink === link.path ? styles.navlinkActive : ''
+        className={`${styles.navlink} ${activeLink === link.path && typeof activeLink !== 'undefined' ? styles.navlinkActive : ''
```
Comment (ankitecd) [inline src/features/common/Layout/UserLayout/UserLayout.tsx:357]: Here you can have condition as `activeLink && activeLink === link.path ? <code> :<code>`

**Review by ankitecd — CHANGES_REQUESTED** [review]
Also document this whole component

**Review by ankitecd — CHANGES_REQUESTED** [review]
- [ ] Here the parent should also be active, if any one child is active.
image

**Review by ankitecd — APPROVED** [review]
Looks good

**Pull request merged**